### PR TITLE
Errors for instanceof operator

### PR DESCRIPTION
**Pull request by hermanventer**
Now give error messages for instanceof operators when the right operand is too abstract.

### CODE REVIEW

**Comment by NTillmann** [comment]
As this change sits on top of, but also includes, previously approved pull requests, it's difficult for me to see what's new. Can you merge the other pull requests and rebase this one on top of them?

**Comment by hermanventer** [comment]
My pull requests always involve only the last commit.

File: src/evaluators/BinaryExpression.js [hunk 1]
```diff
@@ -74,7 +74,7 @@ export function getPureBinaryOperationResultType(
   } else if (op === ">>>" || op === "<<" || op === ">>" || op === "&" || op === "|" || op === "^" ||
              op === "**" || op === "%" || op === "/" || op === "*" || op === "-") {
     return reportErrorIfNotPure(IsToNumberPure, NumberValue);
-  } else if (op === "in") {
+  } else if (op === "in" || op === "instanceof") {
```
Comment (NTillmann) [inline src/evaluators/BinaryExpression.js]: Are plain function values simple? Are bound function values simple? If so, I think the code below would have to peek into `.$BoundTargetFunction` to figure out if this operation is safe.

Comment (hermanventer) [inline src/evaluators/BinaryExpression.js]: Good point. I'll make the isSimple check safer.

Comment (hermanventer) [inline src/evaluators/BinaryExpression.js]: It turns out that it is perfectly safe already. Far too safe, in fact. But making a better alternative to isSimple is best left to a later pull request.

**Pull request closed without merging**